### ISSUE TITLE
[iTMSTransporter] sh: /Applications/Transporter.app/Contents/itms/bin/iTMSTransporter: No such file or directory

### ISSUE THREAD

**Issue by vhsantos26** [actions/runner-images#8127]
### Description

We moved from AWS EC2 Runners to GitHub Runners and, using the current [MacOS latest runner](https://github.com/actions/runner-images/blob/main/images/macos/macos-12-Readme.md), when we try to run the fastlane testflight action to deploy the IPA to the AppStore, we get the following error: 

```
[17:02:23]: [iTMSTransporter] sh: /Applications/Transporter.app/Contents/itms/bin/iTMSTransporter: No such file or directory

[17:02:23]: iTunes Transporter output above ^
[17:02:23]: The call to the iTMSTransporter completed with a non-zero exit status: 127. This indicates a failure.
[17:02:23]: Password contains special characters, which may not be handled properly by iTMSTransporter. If you experience problems uploading to App Store Connect, please consider changing your password to something with only alphanumeric characters.
[17:02:23]: Could not download/upload from App Store Connect! It's probably related to your password or your internet connection.
```

Looks like the Transporter is not installed by default, or we are not using the correct path of it, is there any ideal to solve this?

Fastlane action executed:

```ruby
desc "Testflight Deployment"
lane :deploy do
  testflight(
    api_key: get_app_store_connect_api_key,
    ipa: "./Mobile.ipa",
    apple_id: ENV["APPLE_APP_ID"],
    skip_waiting_for_build_processing: true
  )
end
```

### Platforms affected

- [ ] Azure DevOps
- [ ] GitHub Actions - Standard Runners
- [X] GitHub Actions - Larger Runners

### Runner images affected

- [ ] Ubuntu 20.04
- [ ] Ubuntu 22.04
- [ ] macOS 11
- [X] macOS 12
- [ ] macOS 13
- [ ] Windows Server 2019
- [ ] Windows Server 2022

### Image version and build link

Version: 20230812.3

### Is it regression?

No

### Expected behavior

When the Fastlane action testflight is executed, as the runner have the Transporter installed, the deployment must happen successfully.

### Actual behavior

Getting the error `[iTMSTransporter] sh: /Applications/Transporter.app/Contents/itms/bin/iTMSTransporter: No such file or directory`

### Repro steps

Run the workflow that deploys an IPA to the AppStore Connect like this

```yaml
      - run: bundle exec fastlane deploy
        env:
          APP_STORE_CONNECT_API_KEY: ${{ secrets.APP_STORE_CONNECT_API_KEY }}
          APP_STORE_CONNECT_KEY_ID: ${{ secrets.APP_STORE_CONNECT_KEY_ID }}
          APP_STORE_CONNECT_ISSUER_ID: ${{ secrets.APP_STORE_CONNECT_ISSUER_ID }}
          APPLE_APP_ID: ${{ secrets.APPLE_APP_ID }}
          FASTLANE_ITUNES_TRANSPORTER_USE_SHELL_SCRIPT: 1
          FASTLANE_ITUNES_TRANSPORTER_PATH: /Applications/Transporter.app/Contents/itms
```

Where the Fastlane action is:

```ruby
desc "Testflight Deployment"
lane :deploy do
  testflight(
    api_key: get_app_store_connect_api_key,
    ipa: "./Mobile.ipa",
    apple_id: ENV["APPLE_APP_ID"],
    skip_waiting_for_build_processing: true
  )
end
```

**Comment by Alexey-Ayupov** [comment]
Hi @vhsantos26, we will take a look.

**Comment by sergei-pyshnoi** [comment]
Hello @vhsantos26 . Yes, it is true, we do not install `Transporter.app` to macOS images. After some investigation, I found that is `iTMSTransporter` comes with Xcode, so it can be found in related folder. For example: `/Applications/Xcode_14.2.app/Contents/Developer/usr/bin/iTMSTransporter` (you can replace Xcode version with your needs). Can you try use it and give us feedback?

**Comment by vhsantos26** [comment]
Thanks @sergei-pyshnoi @Alexey-Ayupov, I will try this approach here and let you soon as possible today

**Comment by vhsantos26** [comment]
@sergei-pyshnoi @Alexey-Ayupov, still facing the same issue as we can see here in the log:

```
[15:38:00]: --- Step: testflight ---
[15:38:00]: ------------------------
[15:38:00]: Creating authorization token for App Store Connect API
[15:38:00]: Ready to upload new build to TestFlight (App: ***)...
[15:38:02]: Going to upload updated app to App Store Connect
[15:38:02]: This might take a few minutes. Please don't interrupt the script.
[15:38:02]: [iTMSTransporter] sh: /Applications/Xcode_14.2.app/Contents/Developer/usr/bin/iTMSTransporter/bin/iTMSTransporter: Not a directory

[15:38:02]: iTunes Transporter output above ^
[15:38:02]: The call to the iTMSTransporter completed with a non-zero exit status: 126. This indicates a failure.
[15:38:02]: Password contains special characters, which may not be handled properly by iTMSTransporter. If you experience problems uploading to App Store Connect, please consider changing your password to something with only alphanumeric characters.
[15:38:02]: Could not download/upload from App Store Connect! It's probably related to your password or your internet connection.
```

What I did was to set the following path in the environment: `FASTLANE_ITUNES_TRANSPORTER_PATH: /Applications/Xcode_14.2.app/Contents/Developer/usr/bin/iTMSTransporter`

Should I try with `FASTLANE_ITUNES_TRANSPORTER_PATH: /Applications/Xcode_14.2.app/Contents/Developer/usr/bin/` instead?

**Comment by sergei-pyshnoi** [comment]
> @sergei-pyshnoi @Alexey-Ayupov, still facing the same issue as we can see here in the log:
> 
> ```
> [15:38:00]: --- Step: testflight ---
> [15:38:00]: ------------------------
> [15:38:00]: Creating authorization token for App Store Connect API
> [15:38:00]: Ready to upload new build to TestFlight (App: ***)...
> [15:38:02]: Going to upload updated app to App Store Connect
> [15:38:02]: This might take a few minutes. Please don't interrupt the script.
> [15:38:02]: [iTMSTransporter] sh: /Applications/Xcode_14.2.app/Contents/Developer/usr/bin/iTMSTransporter/bin/iTMSTransporter: Not a directory
> 
> [15:38:02]: iTunes Transporter output above ^
> [15:38:02]: The call to the iTMSTransporter completed with a non-zero exit status: 126. This indicates a failure.
> [15:38:02]: Password contains special characters, which may not be handled properly by iTMSTransporter. If you experience problems uploading to App Store Connect, please consider changing your password to something with only alphanumeric characters.
> [15:38:02]: Could not download/upload from App Store Connect! It's probably related to your password or your internet connection.
> ```
> 
> What I did was to set the following path in the environment: `FASTLANE_ITUNES_TRANSPORTER_PATH: /Applications/Xcode_14.2.app/Contents/Developer/usr/bin/iTMSTransporter`
> 
> Should I try with `FASTLANE_ITUNES_TRANSPORTER_PATH: /Applications/Xcode_14.2.app/Contents/Developer/usr/bin/` instead?

I am not sure how exactly fastlane read path in your case, but in your repro steps `FASTLANE_ITUNES_TRANSPORTER_PATH` is `/Applications/Transporter.app/Contents/itms` and log throw error with `/Applications/Transporter.app/Contents/itms/bin/iTMSTransporter` . Look like the last part of path 'bin/iTMSTransporter' was added by default . I think you should try to pass `/Applications/Xcode_14.2.app/Contents/Developer/usr` as `FASTLANE_ITUNES_TRANSPORTER_PATH` variable.

**Comment by vhsantos26** [comment]
I've set the `FASTLANE_ITUNES_TRANSPORTER_PATH` with `${{ vars.FASTLANE_ITUNES_TRANSPORTER_PATH }} making easier for us to debug, waiting the execution to check here what would be the right value for it

**Comment by vhsantos26** [comment]
@sergei-pyshnoi, thanks for the help, looks like I was incorrect path that I was using and, using your suggestion, I got another error (maybe we are going further? hehe)

```
[20:42:09]: ------------------------
[20:42:09]: --- Step: testflight ---
[20:42:09]: ------------------------
[20:42:09]: Creating authorization token for App Store Connect API
[20:42:09]: Ready to upload new build to TestFlight (App: ***)...
[20:42:10]: Going to upload updated app to App Store Connect
[20:42:10]: This might take a few minutes. Please don't interrupt the script.
[20:42:10]: [iTMSTransporter] An error (-10814) occurred. The operation couldn’t be completed. (OSStatus error -10814.) (null)
```

**Comment by sergei-pyshnoi** [comment]
@vhsantos26 I found [issue](https://github.com/fastlane/fastlane/issues/20650) with similar error. Solution was find in comment [here](https://github.com/fastlane/fastlane/issues/20371#issuecomment-1239093663). I assume that the problem is in the 
`FASTLANE_ITUNES_TRANSPORTER_USE_SHELL_SCRIPT` parameter. Look like it is don`t need if you using Xcode built in transporter.

**Comment by vhsantos26** [comment]
@sergei-pyshnoi, thanks again, I've removed the `FASTLANE_ITUNES_TRANSPORTER_USE_SHELL_SCRIPT` from our workflow file and pushed the change. Let's see the result maybe in an hour after our CI/CD runs 100%.

**Comment by vhsantos26** [comment]
@sergei-pyshnoi Hey, here is the recent execution following the latest change and still failing, another thing now

```
Run bundle exec fastlane deploy
  bundle exec fastlane deploy
  shell: /bin/bash -e {0}
  env:
    MD_APPLE_SDK_ROOT: /Applications/Xcode_14.2.app
    APP_STORE_CONNECT_API_KEY: ***
    APP_STORE_CONNECT_KEY_ID: ***
    APP_STORE_CONNECT_ISSUER_ID: ***
    APPLE_APP_ID: ***
    FASTLANE_ITUNES_TRANSPORTER_PATH: /Applications/Xcode_14.2.app/Contents/Developer/usr
+-------------------------+---------+--------------------------------------------------+
|                                     Used plugins                                     |
+-------------------------+---------+--------------------------------------------------+
| Plugin                  | Version | Action                                           |
+-------------------------+---------+--------------------------------------------------+
| fastlane-plugin-bugsnag | 2.2.1   | send_build_to_bugsnag, upload_symbols_to_bugsnag |
+-------------------------+---------+--------------------------------------------------+

[15:10:50]: Sending anonymous analytics information
[15:10:50]: Learn more at https://docs.fastlane.tools/#metrics
[15:10:50]: No personal or sensitive data is sent.
[15:10:50]: You can disable this by adding `opt_out_usage` at the top of your Fastfile
[15:10:50]: Driving the lane 'deploy' 🚀
[15:10:50]: ----------------------------------------------------------
[15:10:50]: --- Step: Switch to get_app_store_connect_api_key lane ---
[15:10:50]: ----------------------------------------------------------
[15:10:50]: Cruising over to lane 'get_app_store_connect_api_key' 🚖
[15:10:50]: -----------------------------
[15:10:50]: --- Step: ensure_env_vars ---
[15:10:50]: -----------------------------
[15:10:50]: Environment variables 'APP_STORE_CONNECT_API_KEY', 'APP_STORE_CONNECT_ISSUER_ID', 'APP_STORE_CONNECT_KEY_ID' are set!
[15:10:50]: ---------------------------------------
[15:10:50]: --- Step: app_store_connect_api_key ---
[15:10:50]: ---------------------------------------
[15:10:50]: Cruising back to lane 'deploy' 🚘
[15:10:50]: ------------------------
[15:10:50]: --- Step: testflight ---
[15:10:50]: ------------------------
[15:10:50]: Creating authorization token for App Store Connect API
[15:10:50]: Ready to upload new build to TestFlight (App: ***)...
[15:10:52]: Going to upload updated app to App Store Connect
[15:10:52]: This might take a few minutes. Please don't interrupt the script.
[15:10:52]: [iTMSTransporter] sh: /Applications/Xcode_14.2.app/Contents/Developer/usr/java/bin/java: No such file or directory

[15:10:52]: iTunes Transporter output above ^
[15:10:52]: The call to the iTMSTransporter completed with a non-zero exit status: 127. This indicates a failure.
[15:10:52]: The iTMSTransporter Java app was not found at '/Applications/Xcode_14.2.app/Contents/Developer/usr/lib/itmstransporter-launcher.jar'.
[15:10:52]: If you're using Xcode 6, please select the shell script executor by setting the environment variable FASTLANE_ITUNES_TRANSPORTER_USE_SHELL_SCRIPT=1
+---------------+--------+
|      Lane Context      |
+---------------+--------+
| PLATFORM_NAME |        |
| LANE_NAME     | deploy |
+---------------+--------+
[15:10:52]: Error uploading ipa file: 
```

I am trying to find a solution for that but I would like to share here because maybe you guys know something about it

**Comment by vhsantos26** [comment]
I am trying locally as well, I've removed the Transporter app and my Xcode version is 14.3.1 (Build version 14E300c). I've updated the fastlane to 2.210.0 as suggested here.

env set: `export FASTLANE_ITUNES_TRANSPORTER_PATH="/Applications/Xcode.app/Contents/Developer/usr"`

```
 ~/Developer/ump/projects/mobile/ios │ on rc-4.4.0  ls -la /Applications/Xcode.app/Contents/Developer/usr/bin/iTMSTransporter                        
lrwxr-xr-x  1 root  wheel  77 Jul 28 16:38 /Applications/Xcode.app/Contents/Developer/usr/bin/iTMSTransporter -> ../../../SharedFrameworks/ContentDelivery.framework/Resources/TransporterShim
```

but still getting this error locally:

```
[iTMSTransporter] An error (-10814) occurred. The operation couldn’t be completed. (OSStatus error -10814.) (null)
```

**Comment by vhsantos26** [comment]
@sergei-pyshnoi, locally I was able to fix the issue updating the fastlane to 2.210.0. I needed to restart the terminal, I think that was some cache for the envs that I was setting. After that, the deploy worked locally using Xcode 14 and without Transporter (using altool as mentioned before [here](https://github.com/fastlane/fastlane/pull/20631)). I've pushed the changes to the PR, let's see of the CI/CD back to normal
